lib/kafka/connection.rb

```diff
@@ -28,6 +28,10 @@ class Connection
     SOCKET_TIMEOUT = 10
     CONNECT_TIMEOUT = 10
 
+    attr_reader :socket
+    attr_reader :encoder
+    attr_reader :decoder
+
```

Comment:
~~I don't think these attributes are used anywhere...~~ Looks like only `socket` is unused.